src/torrent.cpp

```diff
@@ -1697,7 +1715,7 @@ bool is_downloading_state(int const st)
 
 			for (auto const& ih : m_torrent_file->similar_torrents())
 			{
-				std::shared_ptr<torrent> t = m_ses.find_torrent(ih).lock();
+				std::shared_ptr<torrent> t = m_ses.find_torrent(info_hash_t(ih)).lock();
```

Comment:
does this assume `ih` to be a SHA-1 hash? or would it work if similar-torrents had truncated SHA-256 hashes in it also? Whether similar torrents support v2 hashes or not should be deliberate I think

Reply:
Hybrid torrents are indexed by both their SHA1 and truncated SHA256 hashes, so `find_torrent` will work with both.